src/polynomial.rs

```diff
@@ -102,13 +106,13 @@ impl<S: Scalar> std::ops::Add<Polynomial<S>> for Polynomial<S> {
         if self.degree() >= rhs.degree() {
             let mut x = self.elements;
             for (e, r) in x.iter_mut().zip(rhs.elements.iter()) {
-                *e = *e + r;
+                *e = *e + *r;
```

Comment:
I guess now it allows add assign
```suggestion
                *e += *r;
```